### PR TITLE
Add support for including PowerShell profile

### PR DESCRIPTION
**Pull request by nebula-it**

### CODE REVIEW

File: src/powershell/install.sh
Comment (samruddhikhandale) [inline src/powershell/install.sh:167]: I like the idea of being able to configure the Powershell profile with the Powershell Feature. Can we convert this to a [Feature option](https://containers.dev/implementors/features/#options-property)?

I'd expect it to be a `string` and called something like `powershellProfileURL` which is an optional param set to empty string.

File: src/powershell/install.sh
Comment (samruddhikhandale) [inline src/powershell/install.sh:169]: Any thoughts if we should configure it for all users vs current user? Looks like the current changes are configuring for all users.

I wonder if we need more community feedbacks for this one!

File: src/powershell/install.sh
Comment (samruddhikhandale) [inline src/powershell/install.sh:170]: Also, can you help me understand your motivation behind this change? I have never used Powershell profiles before, and I don't have my own script (hence, this question). 

Do folks like to publish their setting to some location? or are there official scripts available which you like to use?

File: src/powershell/install.sh
Comment (samruddhikhandale) [inline src/powershell/install.sh:169]: > /opt/microsoft/powershell/7/profile.ps1

Looks like ^ would work for Powershell version 7, however, how about other versions? Like 5.1, doesn't seem like `/opt/microsoft/powershell/7/profile.ps1` would work. See https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-5.1

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thank you for taking the time in contributing the changes.

I have left some comments and clarifying questions, thanks!

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:167]: Done :)

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:169]: The dev containers are mostly intended for single users and for shared users so I thought it should be fine to add it at system level. 
Another reason I chose above path is that it remains static v/s a new user might be created inside container at a later stage of dev container setup.
If someone really needs to run pwsh without profile they can launch it using `pwsh -noprofile`. But I'm happy to hear community feedback on this.

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:170]: My PowerShell profile holds a lot of customization (e.g prompt to show git status, custom aliases and functions) and useful utilities like [Completion Predictor](https://github.com/PowerShell/CompletionPredictor) and tab auto completer for lots of tools.

Folks usually publish their profiles to Github. So there are no official scripts etc, just personal profiles usually from Github but could be anywhere on a webserver as long as its publicly accessible.

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:169]: PowerShell 5.1 is not available for Linux and PowerShell 6 is deprecated. Only 7.* is [currently supported](https://learn.microsoft.com/en-us/powershell/scripting/install/powershell-support-lifecycle?view=powershell-7.4), so this should cover most use cases for now. We can expand it later to support other version as well, if needed.

Comment (samruddhikhandale) [reply src/powershell/install.sh:169]: That makes sense, thanks for you explanation.

File: src/powershell/devcontainer-feature.json
Comment (samruddhikhandale) [inline src/powershell/devcontainer-feature.json:26]:
```suggestion
            "description": "Optional (publicly accessible) URL to download PowerShell profile."
```

File: src/powershell/install.sh
Comment (samruddhikhandale) [inline src/powershell/install.sh:167]: Can we add test scenarios to validate these PR changes? Feel free to provide me a script, and I can host it publicly which will be accessible to the test scenario. This is good for maintainability and that way you don't have to maintain/support it long term.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thank you for answering my questions, appreciate it. Left few more comments!

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:167]: Yep for sure, I do not have a public profile atm. For something to test you can use https://github.com/ChrisTitusTech/powershell-profile , the value you will need to use for this parameter is `https://github.com/ChrisTitusTech/powershell-profile/raw/main/setup.ps1`
There are lot more profiles [here](https://github.com/topics/powershell-profile) if you want additional tests.

Comment (samruddhikhandale) [reply src/powershell/install.sh:167]: Thanks, I created a fork at https://github.com/codspace/powershell-profile/tree/main

Can you write tests by using that? Would be great to validate these changes and have some tests added which massages this new Feature option.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Looking for some test coverage, then we are all set.

File: src/powershell/install.sh
Comment (nebula-it) [reply src/powershell/install.sh:167]: Hey @samruddhikhandale,
Apologies for delay in getting back to you, I was bit busy.
Picking this up again, I see three files [here](https://github.com/devcontainers/features/tree/main/test/powershell), but not really clear on how those are being used. Can you please share any docs on how to write test for this or you can guide me through this?

Comment (nebula-it) [reply src/powershell/install.sh:167]: NM, I found the test docs. Will get the test implemented in next day or two.

Comment (nebula-it) [reply src/powershell/install.sh:167]: Alright, I've added test now. I create a simple profile [here](https://github.com/nebula-it/features/blob/patch-2/test/powershell/testProfile.ps1) for testing, as the one I linked is too extensive and not really a good option for testing.

With the current test, if the variable `$env:ProfileLoaded` exists and returns value `True`, that means our profile has been successfully loaded.

File: src/powershell/README.md
Comment (samruddhikhandale) [inline src/powershell/README.md:20]: Once this PR is merged, https://github.com/devcontainers/action will run automatically ; which opens a PR that updates the README file based on `devcontainer-feature.json` and `NOTES.md`.

Hence, can we revert the changes to this file to avoid merge conflicts? Thanks!

File: test/powershell/scenarios.json
Comment (samruddhikhandale) [inline test/powershell/scenarios.json:7]: Would you be able to support and maintain it for longer term? I am worried that the test might break in future if that's not the case.

Can we use https://github.com/codspace/powershell-profile/blob/main/Test-Profile.ps1 instead? Updated the contents to match your test file

File: test/powershell/testProfile.ps1
Comment (samruddhikhandale) [inline test/powershell/testProfile.ps1:1]: Why do we need to add this locally?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: test/powershell/scenarios.json
Comment (nebula-it) [reply test/powershell/scenarios.json:7]: Yeap I can keep it, but using an external repo does not seem like a good solution to me.
So my thought process was that once this was merged, we can point to https://github.com/devcontainers/features/blob/main/test/powershell/testProfile.ps1 so its in same repo

Comment (nebula-it) [reply test/powershell/scenarios.json:7]: Otherwise I can point it to your repo above, let me know which makes more sense to you

Comment (nebula-it) [reply test/powershell/scenarios.json:7]: Oh I see that, above repo you posted is not a personal one. I have updated code to use that.

**Review by samruddhikhandale — APPROVED** [review]
Looks great ⚡ 

Let's wait for test runs to turn 🟢 before we merge it!

**Pull request merged**
